pkg/segment/structs/segstructs.go

```diff
@@ -841,15 +850,35 @@ func CheckUnsupportedFunctions(post *QueryAggregators) error {
 			}
 		}
 
-		// Remove this check once the format command is supported.
-		// Refactor this, if there are more commands that are not supported directly under LetColumnsRequest.
-		if agg.hasLetColumnsRequest() && agg.OutputTransforms.LetColumns.FormatResults != nil {
-			return fmt.Errorf("checkUnsupportedFunctions: using format command is not yet supported")
+		err := checkUnsupportedLetColumnCommand(agg, "FormatResults")
+		if err != nil {
+			return err
+		}
+
+		err = checkUnsupportedLetColumnCommand(agg, "EventCountRequest")
+		if err != nil {
+			return err
 		}
 	}
 
 	return nil
 }
+func checkUnsupportedLetColumnCommand(agg *QueryAggregators, command string) error {
+	if agg.hasLetColumnsRequest() {
+		letColumns := agg.OutputTransforms.LetColumns
+		switch command {
```

Comment:
I think it is better, to keep a global slice or map of `LetColumnsCommand` that we do not support and then loop through that slice or map in the `if` block. 

The code in the `if` block will remain the same and you don't need to pass the command as an argument. 
Through this you will only call the `checkUnsupportedLetColumnCommand` only once in `CheckUnsupportedFunctions` definition.